emulators/mame/Portfile

```
@@ -23,6 +34,80 @@ long_description    The purpose of MAME is to preserve decades of software histo
     the hardware and how it functions. The source code to MAME serves as this \
     documentation.
 
+github.tarball_from archive
+
+#------------------------------------------------------------------------------
+# Patches and Checksums
+#------------------------------------------------------------------------------
+
+# Patch 'language-portuguese_brazil' is only applicable to release 0.226.
+# Fix was submitted, and merged, by Mame maintainers... and will be part of
+# official Mame releases from 0.227 onward. Net-Net, this particular patch can
+# be removed starting with 0.227. Tracked by Mame issue:
+# https://github.com/mamedev/mame/issues/7510
+#
+# Conversely, patch 'scripts-genie-compile-warnings' is not specific to any
+# particular Mame release. Indeed, it is technically not required at all.
+# However, it eliminates spurious warnings due to use of deprecated types
+# and/or calls... and reduces pollution in the build log.
+
+patchfiles          mame-patch-0226-language-portuguese_brazil.diff \
+                    mame-patch-0226-scripts-genie-compile-warnings.diff
+
+checksums           mame-0226.tar.gz \
+                    rmd160  b9e4ae321b7673790d374c63bbe966d1502d6738 \
+                    sha256  7c4c9ec232ba988e65fd29665c9b8e40b5ac3aa9f561eeb107cebbf08ba94baf \
+                    size    196379874 \
+                    mame-patch-0226-language-portuguese_brazil.diff \
+                    rmd160  6eacda9bc43662cc37ad07544fc7e78eb0188b80 \
+                    sha256  63873b094b307d417a0cbc89a23847b876e322466ae809952e8467645cef86f6 \
+                    size    1257 \
+                    mame-patch-0226-scripts-genie-compile-warnings.diff \
+                    rmd160  d45f59b828a0c9cd92d368339852e51b431322e2 \
+                    sha256  4b1f69c8b232b91a667a93d5b904768d4ee082c3aea2051fe8bf279d99010317 \
+                    size    326 \
+                    mame-script-template-launcher.sh \
+                    rmd160  1b752e19fb77b18366f22e86b5578f2682b24c67\
+                    sha256  e7601a1f3fcf9759c42bd5adcbf0d2ba003e91305b8f48dd2e8ed82ff6a5a9d8\
+                    size    280
```

You don't list checksums for files that are in the files directory of the port. MacPorts won't check them. MacPorts only checks checksums for files that are downloaded from master_sites or patch_sites.
```suggestion
                    size    196379874
```